orchagent/fdborch.cpp

```diff
@@ -617,98 +836,324 @@ void FdbOrch::updateVlanMember(const VlanMemberUpdate& update)
-    attr.value.s32 = (type == "dynamic") ? SAI_FDB_ENTRY_TYPE_DYNAMIC : SAI_FDB_ENTRY_TYPE_STATIC;
+    if (fdbData.origin == FDB_ORIGIN_VXLAN_ADVERTIZED)
+    {
+        attr.value.s32 = SAI_FDB_ENTRY_TYPE_STATIC;
+    }
+    else
+    {
+        attr.value.s32 = (fdbData.type == "dynamic") ? SAI_FDB_ENTRY_TYPE_DYNAMIC : SAI_FDB_ENTRY_TYPE_STATIC;
+    }
     attrs.push_back(attr);
 
+    if ((fdbData.origin == FDB_ORIGIN_VXLAN_ADVERTIZED) && (fdbData.type == "dynamic"))
+    {
+        attr.id = SAI_FDB_ENTRY_ATTR_ALLOW_MAC_MOVE;
+        attr.value.booldata = true;
+        attrs.push_back(attr);
+    }
+
     attr.id = SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID;
     attr.value.oid = port.m_bridge_port_id;
     attrs.push_back(attr);
 
-    attr.id = SAI_FDB_ENTRY_ATTR_PACKET_ACTION;
-    attr.value.s32 = SAI_PACKET_ACTION_FORWARD;
-    attrs.push_back(attr);
+    if(fdbData.origin == FDB_ORIGIN_VXLAN_ADVERTIZED)
+    {
+        IpAddress remote = IpAddress(fdbData.remote_ip);
+        sai_ip_address_t ipaddr;
+        if(remote.isV4())
+        {
+            ipaddr.addr_family = SAI_IP_ADDR_FAMILY_IPV4;
+            ipaddr.addr.ip4 = remote.getV4Addr();
+        }
+        else
+        {
+            ipaddr.addr_family = SAI_IP_ADDR_FAMILY_IPV6;
+            memcpy(ipaddr.addr.ip6, remote.getV6Addr(), sizeof(ipaddr.addr.ip6));
+        }
+        attr.id = SAI_FDB_ENTRY_ATTR_ENDPOINT_IP;
+        attr.value.ipaddr = ipaddr;
+        attrs.push_back(attr);
+    }
+    else if(macUpdate 
+            && (oldOrigin == FDB_ORIGIN_VXLAN_ADVERTIZED) 
+            && (fdbData.origin != oldOrigin))
+    {
+        /* origin is changed from Remote-advertized to Local-provisioned
+         * Remove the end-point ip attribute from fdb entry
+         */
+        sai_ip_address_t ipaddr;
+        ipaddr.addr_family = SAI_IP_ADDR_FAMILY_IPV4;
+        ipaddr.addr.ip4 = 0;
+        attr.id = SAI_FDB_ENTRY_ATTR_ENDPOINT_IP;
+        attr.value.ipaddr = ipaddr;
+        attrs.push_back(attr);
+    }
 
-    if (m_entries.count(entry) != 0) // we already have such entries
+    if(macUpdate && (oldOrigin == FDB_ORIGIN_VXLAN_ADVERTIZED)) 
     {
-        removeFdbEntry(entry);
+        if((fdbData.origin != oldOrigin)
+           || ((oldType == "dynamic") && (oldType != fdbData.type)))
+        {
+            attr.id = SAI_FDB_ENTRY_ATTR_ALLOW_MAC_MOVE;
+            attr.value.booldata = false;
+            attrs.push_back(attr);
+        }
     }
 
-    sai_status_t status = sai_fdb_api->create_fdb_entry(&fdb_entry, (uint32_t)attrs.size(), attrs.data());
-    if (status != SAI_STATUS_SUCCESS)
+    
+    if(macUpdate)
+    {
+        SWSS_LOG_NOTICE("MAC-Update FDB %s in %s on from-%s:to-%s from-%s:to-%s origin-%d-to-%d",
+                entry.mac.to_string().c_str(), vlan.m_alias.c_str(), oldPort.m_alias.c_str(),
+                port_name.c_str(), oldType.c_str(), fdbData.type.c_str(),
+                oldOrigin, fdbData.origin);
+        for(auto itr : attrs)
```

Comment:
space after `for `and few of the `if `statements in PR

Reply:
resolved in last commit